### PR TITLE
[NT-1939] Navigation to Comment Replies

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

- Allows us to navigate to comment replies by tapping on a root comment.
- Displays _view replies_ on a comment when it has replies.

# 🤔 Why

Continuation of the work we're doing to bring threaded comments to the native apps, this is necessary to allow us to navigate to the replies.

# 🛠 How

- Updated our data source to allow to grab a comment at the selected `IndexPath`.
- Added inputs and outputs to `CommentsViewModel` and added tests.
- Added an empty `CommentRepliesViewController` to navigate to.
- Conditionally added the view to `CommentCell` when the associated `Comment` has replies.

# 👀 See

| Before 🐛 | After 🦋 |
| --- | --- |
| image | image |

# ✅ Acceptance criteria

- [x] Navigate to comments, tap on a comment, an empty table view should be pushed onto the navigation stack.
- [x] Tapping on comments that are in error or deleted should not navigate to the replies VC.
- [x] The view indicating that a comment has replies should only be shown for non-errored, non-deleted comments and for comments that have replies.

### CODE REVIEW

File: Kickstarter-iOS/Views/Cells/CommentCell.swift [hunk 1]
```diff
@@ -7,13 +7,16 @@ final class CommentCell: UITableViewCell, ValueCell {
...
+  private lazy var bottomRowStackView: UIStackView = { UIStackView(frame: .zero) }()
   private lazy var commentCellHeaderStackView: CommentCellHeaderStackView = {
     CommentCellHeaderStackView(frame: .zero)
   }()
 
   private lazy var flagButton = { UIButton(frame: .zero) }()
   private lazy var replyButton = { UIButton(frame: .zero) }()
+  private lazy var viewRepliesStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private lazy var viewRepliesLabel: UILabel = { UILabel(frame: .zero) }()
+  private lazy var viewRepliesIconImageView: UIImageView = { UIImageView(frame: .zero) }()
```
Comment (singhhari) [inline Kickstarter-iOS/Views/Cells/CommentCell.swift:19]: alphabetize

File: Library/ViewModels/CommentCellViewModel.swift [hunk 2]
```diff
@@ -11,23 +11,26 @@ public protocol CommentCellViewModelInputs {
...
-  /// Emits text  relative time the comment was posted
+  /// Emits text  relative time the comment was posted.
   var postTime: Signal<String, Never> { get }
 
-  /// Emits a Bool determining if the reply and flag buttons in the bottomColumnStackView are hidden
+  /// Emits a Bool determining if the reply and flag buttons in the bottomColumnStackView are hidden.
   var replyButtonIsHidden: Signal<Bool, Never> { get }
+
+  /// Emits whether or not the view replies stack view is hidden.
+  var viewRepliesStackViewHidden: Signal<Bool, Never> { get }
```
Comment (singhhari) [inline Library/ViewModels/CommentCellViewModel.swift:33]: If this is emitting a Bool we should probably follow the Swift convention of the name beginning with `is`

File: Library/ViewModels/CommentsViewModel.swift [hunk 3]
```diff
@@ -4,10 +4,13 @@ import ReactiveExtensions
 import ReactiveSwift
 
 public protocol CommentsViewModelInputs {
-  /// Call after instantiating the view controller
+  /// Call with a `Comment` when it is selected.
+  func didSelectComment(_ comment: Comment)
```
Comment (singhhari) [inline Library/ViewModels/CommentsViewModel.swift:8]: alphabetize

**Review by singhhari — COMMENTED** [review]
looks good. left some small formatting comments.

File: Kickstarter-iOS/Views/Cells/CommentCell.swift (hunk 1 above)
Comment (justinswart) [reply Kickstarter-iOS/Views/Cells/CommentCell.swift:19]: oooh snap! 😅

**Review by singhhari — APPROVED** [review]
lgtm

**Review by moderateepheezy — APPROVED** [review]

**Pull request merged**